src/models/simple_metabolism_model.py

```diff
@@ -65,43 +65,38 @@ def run(self, carb_amount, insulin_amount=np.nan, num_hours=8):
         if num_hours > 24:
             raise ValueError("Number of hours for simulation can't be more than 24.")
 
-        # create a time series
-        t = get_timeseries(num_hours, five_min=False)
-        t_5min = get_timeseries(num_hours, five_min=True)
+        if carb_amount <= 0 and insulin_amount <= 0:
+            raise ValueError("Insulin or carbs must be greater than zero.")
 
         # if insulin amount is not given,
         # calculate carb amount like a bolus calculator
         if np.isnan(insulin_amount):
             insulin_amount = carb_amount / self._cir  # insulin amount
 
-        # insulin model
-        if insulin_amount != 0:
-
-            # model constants
-            i_t, iob_t = self.insulin_model.run(t, insulin_amount=insulin_amount)
+        sim_data_len = num_hours * MINUTES_PER_HOUR
+        if five_min:
+            sim_data_len = int(sim_data_len / 5)
 
-            ie_5min = i_t[t_5min]
-            iob_5min = iob_t[t_5min]
-            decrease_due_to_insulin = np.append(0, ie_5min[1:] - ie_5min[:-1])
+        # Init arrays to return
+        combined_delta_bg = np.zeros(sim_data_len)
+        iob = np.zeros(sim_data_len)
 
-        else:
-            decrease_due_to_insulin = np.zeros(len(t_5min))
-            iob_5min = np.zeros(len(t_5min))
+        # insulin model
+        if insulin_amount > 0:
+            t_min, bg_delta_insulin, bg, iob = self.insulin_model.run(num_hours,
+                                                               insulin_amount=insulin_amount,
+                                                               five_min=five_min)
+            combined_delta_bg += bg_delta_insulin
 
         # carb model
         if carb_amount > 0:
-
-            c_t = self.carb_model.run(t, carb_amount=carb_amount)
-            ce_5min = c_t[t_5min]
-            increase_due_to_carbs = np.append(0, ce_5min[1:] - ce_5min[:-1])
-
-        else:
-            increase_due_to_carbs = np.zeros(len(t_5min))
-
-        net_change_in_bg = decrease_due_to_insulin + increase_due_to_carbs
+            t_min, bg_delta_carb_minutes, bg = self.carb_model.run(num_hours,
+                                         carb_amount=carb_amount,
+                                         five_min=five_min)
+            combined_delta_bg += bg_delta_carb_minutes
 
         # +CS - Why are we returning the carb and insulin amt?
-        return net_change_in_bg, t_5min, carb_amount, insulin_amount, iob_5min
+        return combined_delta_bg, t_min, carb_amount, insulin_amount, iob
 
     def get_iob_from_sbr(self, sbr_actual):
```

Comment:
I would change this to: Compute insulin on board due to the assumption that the schedule basal rate (sbr) has been active for at least <insulin duration> hours (6 hours in this case)prior to the start of the simulation. The effect of the insulin due to the sbr prior to the start of the simulation results in an initial amount of insulin on board every 5 minutes over the <insulin duration>.

Reply:
That's more descriptive, will do. 6 hours, not 8?

Reply:
Oh I see, you're referring to 6 due to DIA. It's coded as 8 in your original code, but we could require a DIA when instantiating the SimpleMetabolismModel class and use that value.